halo2_gadgets/src/ecc/chip/mul.rs

```diff
@@ -425,6 +425,10 @@ impl<F: FieldExt> Deref for Z<F> {
 }
 
 fn decompose_for_scalar_mul(scalar: Option<&pallas::Base>) -> Vec<Option<bool>> {
+    construct_uint! {
+        struct U256(4);
+    }
+
```

Comment:
I'm surprised that this is only needed locally! Would it make sense at all for this to be a top-level declaration?

Reply:
We only require this because we need to perform a non-reducing addition internal to the decomposition. It's a specific artifact of how variable-base scalar multiplication is implemented for this chip; it does not need to be part of the public API, or even be interoperable with other uses of it elsewhere in `halo2_gadgets`. That being said, if we do ever find ourselves in need of such interoperability, we can refactor at that time.